core/adapter/activity_handler.go

```diff
@@ -0,0 +1,28 @@
+package adapter
+
+
+import (
+	"github.com/infracloudio/msbotbuilder-go/schema"
+)
+
+type ActivityHandler interface {
+	OnMessage(turn *TurnContext) interface{}
```

Comment:
Can we add one more function here to just verify that it won't fail if we add a single or no handler in application?